spec/lib/audit/catalog_to_moab_instance_spec.rb

```diff
@@ -109,6 +109,16 @@
         expect(moab).not_to receive(:current_version_id)
         c2m.check_catalog_version
       end
+      context 'DB transaction handling' do
+        it 'on transaction failure, completes without raising error, removes PC_STATUS_CHANGED result code' do
+          allow(Moab::StorageObject).to receive(:new).with(druid, instance_of(String)).and_return(nil)
+          allow(pres_copy).to receive(:save!).and_raise(ActiveRecord::ConnectionTimeoutError)
+          c2m.check_catalog_version
+          expect(c2m.results.result_array).to include(a_hash_including(AuditResults::MOAB_NOT_FOUND))
+          expect(c2m.results.result_array).not_to include(a_hash_including(AuditResults::PC_STATUS_CHANGED))
+          expect(pres_copy.reload.status).not_to eq PreservedCopy::ONLINE_MOAB_NOT_FOUND_STATUS
```

Comment:
i felt a little strange setting an expectation that the status would be wrong, but this is what should happen.  i mostly wanted to make it clear in the test why the result code was getting removed, which i think is the more important part of the test.